homeassistant/components/fivem/__init__.py

```diff
@@ -0,0 +1,185 @@
+from homeassistant.const import CONF_HOST, CONF_NAME, CONF_PORT, Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers.entity import DeviceInfo, EntityDescription
+from homeassistant.helpers.update_coordinator import (
+    CoordinatorEntity,
+    DataUpdateCoordinator,
+    UpdateFailed,
+)
+
+from .const import (
+    ATTR_PLAYERS_LIST,
+    ATTR_RESOURCES_LIST,
+    DOMAIN,
+    MANUFACTURER,
+    NAME_PLAYERS_MAX,
+    NAME_PLAYERS_ONLINE,
+    NAME_RESOURCES,
+    NAME_STATUS,
+    SCAN_INTERVAL,
+)
+
+PLATFORMS: list[Platform] = [Platform.BINARY_SENSOR, Platform.SENSOR]
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up FiveM from a config entry."""
+    _LOGGER.debug(
+        "Create FiveM server instance for '%s:%s'",
+        entry.data[CONF_HOST],
+        entry.data[CONF_PORT],
+    )
+
+    try:
+        coordinator = FiveMDataUpdateCoordinator(hass, entry.data, entry.entry_id)
+        await coordinator.initialize()
+    except FiveMServerOfflineError as err:
+        raise ConfigEntryNotReady from err
+
+    await coordinator.async_config_entry_first_refresh()
+    entry.async_on_unload(entry.add_update_listener(update_listener))
+
+    hass.data.setdefault(DOMAIN, {})[entry.entry_id] = coordinator
+
+    hass.config_entries.async_setup_platforms(entry, PLATFORMS)
+
+    return True
+
+
+async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Unload a config entry."""
+    if unload_ok := await hass.config_entries.async_unload_platforms(entry, PLATFORMS):
+        hass.data[DOMAIN].pop(entry.entry_id)
+
+    return unload_ok
+
+
+async def update_listener(hass: HomeAssistant, entry: ConfigEntry) -> None:
+    """Update listener."""
+    await hass.config_entries.async_reload(entry.entry_id)
+
+
+class FiveMDataUpdateCoordinator(DataUpdateCoordinator[dict[str, Any]]):
+    """Class to manage fetching FiveM data."""
+
+    def __init__(self, hass: HomeAssistant, config_data, unique_id: str) -> None:
+        """Initialize server instance."""
+        self._hass = hass
+
+        self.unique_id = unique_id
+        self.server = None
+        self.version = None
+        self.gamename: str | None = None
+
+        self.server_name = config_data[CONF_NAME]
+        self.host = config_data[CONF_HOST]
+        self.port = config_data[CONF_PORT]
+        self.online = False
+
+        self._fivem = FiveM(self.host, self.port)
+
+        update_interval = timedelta(seconds=SCAN_INTERVAL)
+        _LOGGER.debug("Data will be updated every %s", update_interval)
+
+        super().__init__(hass, _LOGGER, name=DOMAIN, update_interval=update_interval)
+
+    async def initialize(self) -> None:
+        """Initialize the FiveM server."""
+        info = await self._fivem.get_info_raw()
+        self.server = info.get("server")
```

Comment:
```suggestion
        self.server = info["server"]
```

I am not fully sure what this server does and why you are initializing it in the DataUpdateCoordinator (instead of `__init__` and just passing it), however are you not sure if the values are present? Otherwise use [] instead of .get()

Reply:
I am initializing it in the DataUpdateCoordinator, because I cannot call async methods in the `__init__` method. I replaced the get() with []